### PR TITLE
Allow for local salt content

### PR DESCRIPTION
**Pull request by YakDriver**

### CODE REVIEW

File: src/watchmaker/utils/__init__.py [hunk 1]
```diff
@@ -58,3 +59,32 @@ def urlopen_retry(uri):
...
+            Whether to delete destination prior to copy.
+            (*Default*: ``False``)
+
+        Returns:
+            :obj:`bool`:
+                ``True`` if the copy was attempted. ``False`` if not.
+
+    """
+    if force and os.path.exists(dst):
+        shutil.rmtree(dst)
```
Comment (YakDriver) [inline src/watchmaker/utils/__init__.py]: Tortured logic removed here (`ignore` stuff).

File: src/watchmaker/workers/salt.py [hunk 2]
```diff
@@ -255,6 +258,21 @@ def _build_salt_formula(self, extract_dir):
                 to_directory=extract_dir
             )
 
+        bundled_content = os.sep.join(
+            (static.__path__[0], 'salt', 'content')
+        )
```
Comment (YakDriver) [inline src/watchmaker/workers/salt.py]: Removed a check for whether `bundled_content` directory exists.

File: src/watchmaker/workers/salt.py [hunk 3]
```diff
@@ -255,6 +258,21 @@ def _build_salt_formula(self, extract_dir):
...
+        for subdir in next(os.walk(bundled_content))[1]:
+            if not subdir.startswith('.'):
```
Comment (YakDriver) [inline src/watchmaker/workers/salt.py]: Re-wrote to remove `copytree()` call from `if`.

**Comment by YakDriver** [comment]
@lorengordon Your previous review comment was about getting rid of flags for force, which I've done.

File: src/watchmaker/utils/__init__.py [hunk 4]
```diff
@@ -58,3 +59,32 @@ def urlopen_retry(uri):
...
+
+    shutil.copytree(src, dst, **kwargs)
+    return True
```
Comment (lorengordon) [inline src/watchmaker/utils/__init__.py]: We can remove this return now

File: src/watchmaker/workers/salt.py [hunk 5]
```diff
@@ -241,7 +240,9 @@ def _get_formulas_conf(self):
         ]
 
     def _build_salt_formula(self, extract_dir):
-        if self.salt_content:
+
```
Comment (lorengordon) [inline src/watchmaker/workers/salt.py]: The spaces around this `if` seem extraneous, and throw off the diff. Let's minimize the changes please

File: src/watchmaker/workers/salt.py [hunk 6]
```diff
@@ -255,6 +256,21 @@ def _build_salt_formula(self, extract_dir):
                 to_directory=extract_dir
             )
 
+        bundled_content = os.sep.join(
+            (static.__path__[0], 'salt', 'content')
+        )
+        for subdir in next(os.walk(bundled_content))[1]:
+            if not subdir.startswith('.') and \
```
Comment (lorengordon) [inline src/watchmaker/workers/salt.py]: The spacing might be cleaner by wrapping the conditions in parens. What do you think? 

```suggestion
            if (
                not subdir.startswith('.') and
                not os.path.exists(os.sep.join((extract_dir, subdir)))
            ):
```

File: tests/test_saltworker.py [hunk 7]
```diff
@@ -53,3 +62,236 @@ def test_valid_environment(saltworker_client):
...
+        encoding="utf-8"
+    )
+    mock_safe.assert_called_with(
+        saltworker_win.salt_conf,
+        mock_codec.return_value.__enter__(),
+        default_flow_style=False
+    )
+
+
+@patch('os.path.exists', autospec=True)
```
Comment (lorengordon) [inline tests/test_saltworker.py]: These tests of watchmaker.utils should probably go in another file, `test_utils.py`

**Comment by YakDriver** [comment]
@lorengordon made changes and ran Terrafirm to test bundled and partially user-provided content. Everything passed.

**Pull request merged**